### PR TITLE
Publisher#flatMapMerge recover from mapped Publisher exception

### PR DESCRIPTION
**Pull request by Scottmitch**
Motivation:
Publisher#flatMapMerge does not attempt to recover from throwing in
onNext from a mapped publisher. In this case we should attempt to
replenish demand incase there is a recoverWith or similar operator
applied to the mapped publisher, or else we may deadlock.

Modifications:
- FlatMapSubscriber should try to replenish demand even if onNext throws
an exception

Result:
Publisher#flatMapMerge is compatible with recoverWith style operators on
the mapped publishers.

### CODE REVIEW

**Comment by Scottmitch** [comment]
build failure attributed to https://github.com/apple/servicetalk/issues/1192 and https://github.com/apple/servicetalk/issues/1219

test this please

File: servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java [hunk 1]
```diff
@@ -96,6 +105,43 @@ public static void afterClass() throws Exception {
...
+
+        Processor<Integer, Integer> processor = newPublisherProcessor();
+        toSource(fromSource(processor).flatMapMergeDelayError(i -> from(i + 10).recoverWith(cause ->
+                from(i + 20).concat(failed(cause))))).subscribe(mockSubscriber);
+
+        latchOnSubscribe.await();
+        processor.onNext(1);
+        assertThat(results.take(), is(11));
+        processor.onError(DELIBERATE_EXCEPTION);
+        assertThat(results.take(), is(21));
```
Comment (idelpivnitskiy) [inline servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java:140]: The ordering of events in this test is a bit confusing. `results` queue already has `21` before `processor.onError(DELIBERATE_EXCEPTION)` is invoked. Consider moving this check up if this is expected.
Why the concatenated `cause` is not delivered to `mockSubscriber`?

Comment (idelpivnitskiy) [reply servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java:140]: oh, bcz we delay error.

Comment (idelpivnitskiy) [reply servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java:140]: WDYT about this ordering of events?
```java
        latchOnSubscribe.await();
        processor.onNext(1);
        assertThat(results.take(), is(11));
        assertThat(results.take(), is(21));
        assertThat(causeRef.get(), is(nullValue()));
        processor.onComplete();
        latchOnError.await();
        final Throwable t = causeRef.get();
        assertThat(t, is(instanceOf(CompositeException.class)));
        assertThat(t.getCause(), is(DELIBERATE_EXCEPTION));
```

Comment (Scottmitch) [reply servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/PublisherFlatMapMergeTest.java:140]: I adjusted slightly but sgtm!

**Review by idelpivnitskiy — APPROVED** [review]
Good catch 🎖️

**Pull request merged**